modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/search/insights/view.jsp

```diff
@@ -72,11 +80,39 @@ SearchInsightsDisplayContext searchInsightsDisplayContext = (SearchInsightsDispl
 					persistState="<%= true %>"
 					title="response-string"
 				>
-					<code>
-						<%= HtmlUtil.escape(searchInsightsDisplayContext.getResponseString()) %>
-					</code>
+					<clay:button
+						displayType="secondary"
+						icon="copy"
+						label="copy-to-clipboard"
+						onClick='<%= liferayPortletResponse.getNamespace() + "copyToClipboard('insightsResponse');" %>'
+						small="<%= true %>"
+					/>
+
+					<textarea readonly id="<portlet:namespace />insightsResponse"><%= HtmlUtil.escape(searchInsightsDisplayContext.getResponseString()) %></textarea>
 				</liferay-ui:panel>
 			</liferay-ui:panel-container>
 		</div>
 	</c:otherwise>
-</c:choose>
\ No newline at end of file
+</c:choose>
+
+<aui:script>
+	function <portlet:namespace />copyToClipboard(id) {
+		const text = document.getElementById('<portlet:namespace />' + id).value;
```

Comment:
Instead of hardcoding the namespace in the function, I think it'd be more readable relying on the `id` parameter entirely. That way the mapping between the uses connect more:
-  `copyToClipboard('<portlet:namespace />insightsResponse');` 
- `<textarea readonly id="<portlet:namespace />insightsResponse">`

```suggestion
	function <portlet:namespace />copyToClipboard(id) {
		const text = document.getElementById(id).value;
```